sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -282,6 +320,12 @@ public void finish(@Nullable SpanStatus status) {
     }
   }
 
+  private @Nullable Date findLatestChildTimestamp() {
+    final List<Span> spans = new ArrayList<>(this.children);
+    Collections.sort(spans, spanByTimestampComparator);
```

Comment:
not sure I got it, wdym by  "sorting before traversing"?

Reply:
I meant, what's more expensive?
Sorting the list using a comparator as you did or just traversing the array and doing an `if` check it in a for loop?
I believe the comparator will be more expensive since it needs to move all the items around.